server/src/main/java/org/elasticsearch/search/rank/rrf/RRFRankDoc.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.rank.rrf;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.search.rank.RankDoc;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * {@code RRFRankDoc} supports additional ranking information
+ * required for RRF.
+ */
+public class RRFRankDoc extends RankDoc {
```

Comment:
Similar to the comment I left on RankDoc, I am hoping that we can do without this class

Reply:
My response is similar to the one for RankDoc as well :) . I do think this could be replaced by FieldDoc, but given the complexity would be best served in a follow up PR.